src/auth.rs

```diff
@@ -28,13 +29,25 @@ pub fn authenticate(
                 "found a matching key: {}",
                 key.fingerprint(Default::default())
             ))?;
-            return sign_and_verify(key, agent);
+            // Allow sign_and_verify() to return RemoteFailure (key not loaded / present),
+            // and try the next configured key
+            match sign_and_verify(&key, &mut agent) {
+                Ok(res) => return Ok(res),
+                Err(e) => {
+                    if let Some(SACError::RemoteFailure) = e.downcast_ref::<SACError>() {
+                        log.debug(format!("SSHAgent: RemoteFailure; trying next key"))?;
```

Comment:
Since the `Log` trait accepts anything that implements Display, you can send a raw `&str` here and simply get rid of the `format!()` macro invocation in the middle and it will work the same.